hnn_core/tests/test_general_optimization.py

```diff
@@ -0,0 +1,96 @@
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+from hnn_core import jones_2009_model, simulate_dipole
+
+from hnn_core.optimization import Optimizer
+
+import pytest
+
+
+@pytest.mark.parametrize("solver", ['bayesian', 'cobyla'])
+def test_optimize_evoked(solver):
+    """Test optimization routines for evoked drives in a reduced network."""
+
+    tstop = 10.
+    n_trials = 1
+
+    # simulate a dipole to establish ground-truth drive parameters
+    net_orig = jones_2009_model()
+    net_orig._N_pyr_x = 3
+    net_orig._N_pyr_y = 3
```

Comment:
wonder if we should allow a tuple `network_size=(N_pyr_x, N_pyr_y)` as arguments to `Network` class and in turn to the `jones_2009_model` etc to avoid such issues.

Reply:
So changing `net._N_pyr_x` actually does produce a reduced network but you were right in that something really weird was going on, max_iter was still getting hard coded in the `__init__`...🤦‍♀️